xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -701,9 +704,21 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterDateProperty)
-  ## TODO: The date filter should be smarter than this..
-  #set ($whereSql = "${whereSql} and upper(str(${safe_tableAlias}.value)) like upper(?)")
-  #set ($discard = $whereParams.add("%$filterValue%"))
+  #if ($safe_tableAlias.startsWith('doc.'))
+    #set ($dateProperty = $safe_tableAlias)
+  #else
+    #set ($dateProperty = "${safe_tableAlias}.value")
+  #end
+  #set ($timestamps = $filterValue.split('-'))
+  #if ($filterValue.contains('-') &amp;&amp; $timestamps.size() == 2)
+    #set ($whereSql = "${whereSql} and $dateProperty >= ? and $dateProperty <= ?")
+    #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[0]))))
+    #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[1]))))
+  #else
+    ## Fall back to the older string matching
+    #set ($whereSql = "${whereSql} and upper($dateProperty) like upper(?)")
+    #set ($discard = $whereParams.add($filterValue))
```

Comment:
I forgot to add the percentages for the `like`. `$dateProperty` is because we want to support two cases.